Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -110,6 +125,15 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
             break
         }
     }
+    
+    @objc fileprivate func refreshBookmarks() {
+        profile.syncManager.mirrorBookmarks().upon { (_) in
+            DispatchQueue.main.async {
+                self.loadData()
```

Comment:
What happens if you pull to refresh when you're in a subfolder?

What happens if that subfolder is deleted as a result of the sync?

@joeyg, @jhugman — did you test this?

Reply:
@rnewman I tested deleting the folder that is selected and all that is displayed is the Header linking to the parent folder. I have not yet tested deleting the parent folder of the current folder - I can take a look that when I have some time.